src/mbgl/text/placement.hpp

```diff
@@ -87,11 +87,18 @@ class CollisionGroups {
 
 class BucketPlacementParameters {
 public:
+    friend bool operator<(const BucketPlacementParameters& lhs, const BucketPlacementParameters& rhs) {
+        return lhs.sortKey < rhs.sortKey;
+    }
+    SymbolBucket& bucket;
     const RenderTile& tile;
     const mat4& projMatrix;
     std::string sourceId;
     std::shared_ptr<FeatureIndex> featureIndex;
     bool showCollisionBoxes;
+    float sortKey;
+    size_t symbolInstanceStart;
+    size_t symbolInstanceEnd;
```

Comment:
could the newly added fields be `SymbolBucket` properties?

Reply:
These properties come from `SortKeyRange`. A bucket can have many `SortKeyRange`s